File: FFinder/app/src/main/java/com/ffinder/android/enums/SearchResult.java
package com.ffinder.android.enums;

import android.content.Context;
import com.ffinder.android.R;

/**
 * Created by SiongLeng on 3/9/2016.
 */
public enum SearchResult {
    Normal,
    ErrorNoLink, ErrorTimeoutUnknownReason,
    ErrorLocationDisabled,
    ErrorTimeoutNoConnection, ErrorUserBlocked;


    public String getMessage(Context context){
        switch (this){
            case ErrorNoLink:
                return context.getString(R.string.error_no_link_msg);
            case ErrorTimeoutUnknownReason:
                return context.getString(R.string.error_timeout_msg);
            case ErrorLocationDisabled:
                return context.getString(R.string.error_location_disabled_msg);
            case ErrorTimeoutNoConnection:
                return context.getString(R.string.error_timeout_no_connection_msg);
            case ErrorUserBlocked:
                return context.getString(R.string.error_user_blocked_msg);
        }


        return this.name();
    }

    public boolean isError(){
        return (this.name().startsWith("Error"));
    }

    public boolean errorTriggeredAutoNotification(){
        return this == ErrorTimeoutUnknownReason;
    }

}

File: FFfinder_mac/FFfinder/FFfinder/SearchResult.swift
//
//  SearchResult.swift
//  FFfinder
//
//  Created by Siong Leng Ho on 16/10/16.
//  Copyright © 2016 Lightweight Studio Apps. All rights reserved.
//

import Foundation
import EVReflection

enum SearchResult:String, EVRawString{
    
    case Normal = "Normal"
    case ErrorNoLink = "ErrorNoLink"
    case ErrorTimeoutUnknownReason = "ErrorTimeoutUnknownReason"
    case ErrorTimeoutLocationDisabled = "ErrorTimeoutLocationDisabled"
    case ErrorTimeoutNoConnection = "ErrorTimeoutNoConnection"
    case ErrorUserBlocked = "ErrorUserBlocked"
    
    public static func getMessage(_ result:SearchResult) -> String{
        switch result {
        case .ErrorNoLink:
            return "error_no_link_msg".localized
        case .ErrorTimeoutUnknownReason:
            return "error_timeout_msg".localized
        case .ErrorTimeoutLocationDisabled:
            return "error_timeout_msg".localized
        case .ErrorTimeoutNoConnection:
            return "error_timeout_no_connection_msg".localized
        case .ErrorUserBlocked:
            return "error_user_blocked_msg".localized
        default:
            return ""
        }
    }
    
    public func isError() -> Bool{
        return self.rawValue.contains("Error")
    }
    
    
    public func errorTriggeredAutoNotification() -> Bool{
        return self == SearchResult.ErrorTimeoutUnknownReason
    }
   
    
    
    
}
